indexing-service/src/main/java/io/druid/indexing/overlord/setup/WorkerBehaviorConfig.java

```diff
@@ -19,85 +19,19 @@
 
 package io.druid.indexing.overlord.setup;
 
-import com.fasterxml.jackson.annotation.JsonCreator;
-import com.fasterxml.jackson.annotation.JsonProperty;
-import io.druid.indexing.overlord.autoscaling.AutoScaler;
-import io.druid.indexing.overlord.autoscaling.NoopAutoScaler;
-
-/**
- */
-public class WorkerBehaviorConfig
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import io.druid.guice.annotations.ExtensionPoint;
+
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type", defaultImpl = DefaultWorkerBehaviorConfig.class)
+@JsonSubTypes(value = {
+    @JsonSubTypes.Type(name = "default", value = DefaultWorkerBehaviorConfig.class)
+})
+@ExtensionPoint
+public interface WorkerBehaviorConfig
```

Comment:
What's the motivation behind removing getAutoScaler from this interface, and then moving it to one specific implementation (DefaultWorkerBehaviorConfig), and then making many other things (through getWorkerBehaviorConfig) depend on that specific implementation?

It seems like a violation of good interface design, but I assume you had some reason for it.

Reply:
Maybe an extension you're implementing makes more sense with this design?

If that's the case could you consider adding the rationale to a common on something -- perhaps on getWorkerBehaviorConfig or on DefaultWorkerBehaviorConfig.

Reply:
Added comment

Reply:
Thanks, it looks good to me.